### DISCUSSION TITLE
stucked with sysex

### DISCUSSION THREAD

**Discussion by eqcq** [FortySevenEffects/arduino_midi_library#322] [Q&A] [answered]
Hi,

 a midi device I've made, receive and read sysex message with this function :

```c++
void OnMidiSysEx(byte* pData, unsigned length) {
  std::string myString((char*)pData, length);
  String message = myString.c_str();
  ChaineMemoire = message;
}
```

Everything is ok with ios app I've developped to send message to my device

now I wanted to send sysex with another device I made using this code :

```c++
#include <MIDI.h>
MIDI_CREATE_DEFAULT_INSTANCE();

void sendsysex() {
   uint8_t sysex[] = {0xF0, 0x65, 0x73, 0x73, 0x61, 0x69, 0x0A, 0xF7};
   MIDI.sendSysEx(8, sysex, false);
}

void setup() {
  MIDI.begin();
  // put your setup code here, to run once:
}

void loop() {
  sendsysex();
  delay(5000);
}
```

and the message don't seems to be received. what do I do wrong ?
 THX !

**Comment by franky47** [comment, marked as answer]
Your message does contain the SysEx boundaries (0xF0 and 0xF7), so you'll need to pass `true` to the last argument of `sendSysEx`:
```c++
void sendsysex() {
   uint8_t sysex[] = {0xF0, 0x65, 0x73, 0x73, 0x61, 0x69, 0x0A, 0xF7};
   MIDI.sendSysEx(8, sysex, true);
}
```

**Comment by eqcq** [comment]
HI,
same thing, message doesn’t seem to be read.

I’ve tested wiring, cables etc. everything is fine.
thx !

**Reply by franky47** [reply]
Could you use a MIDI monitor to inspect what is sent by the iOS app (and understood by your receiving device), and what is sent by your emitting device, and find the difference?

**Comment by eqcq** [comment]
ok merci !
c’etait un faux contact …

**Reply by franky47** [reply]
Le coup classique. 😅

**Comment by eqcq** [comment]
Voila 😂.Le 19 févr. 2023 à 21:55, François Best ***@***.***> a écrit :﻿
Le coup classique. 😅
